interface/main/messages/messages.php

```diff
@@ -35,599 +37,670 @@
-        // Add a new message; it's treated as a new note in Patient Notes.
-        $note = $_POST['note'];
-        $noteid = $_POST['noteid'];
-        $form_note_type = $_POST['form_note_type'];
-        $form_message_status = $_POST['form_message_status'];
-        $reply_to = $_POST['reply_to'];
-        $assigned_to_list = explode(';', $_POST['assigned_to']);
-        foreach($assigned_to_list as $assigned_to){
-          if ($noteid && $assigned_to != '-patient-') {
-            updatePnote($noteid, $note, $form_note_type, $assigned_to, $form_message_status);
-            $noteid = '';
-          }
-          else {
-            if($noteid && $assigned_to == '-patient-'){
-              // When $assigned_to == '-patient-' we don't update the current note, but
-              // instead create a new one with the current note's body prepended and
-              // attributed to the patient.  This seems to be all for the patient portal.
-              $row = getPnoteById($noteid);
-              if (! $row) die("getPnoteById() did not find id '".text($noteid)."'");
-              $pres = sqlQuery("SELECT lname, fname " .
-                "FROM patient_data WHERE pid = ?", array($reply_to) );
-              $patientname = $pres['lname'] . ", " . $pres['fname'];
-              $note .= "\n\n$patientname on ".$row['date']." wrote:\n\n";
-              $note .= $row['body'];
+require_once("$srcdir/formatting.inc.php");
+require_once("$srcdir/MedEx/API.php");
+
+$MedEx = new MedExApi\MedEx('MedExBank.com');
+$logged_in = $MedEx->login();
+
+?><html>
+  <head>
+    <title><?php echo xlt('Message Center'); ?></title>
+    <link rel="stylesheet" href="<?php echo $css_header;?>" type="text/css">
+    <link rel="stylesheet" href="<?php echo $GLOBALS['assets_static_relative'] ?>/jquery-ui-1-11-4/themes/excite-bike/jquery-ui.css">
+    <link rel="stylesheet" href="<?php echo $GLOBALS['assets_static_relative'] ?>/bootstrap-3-3-4/dist/css/bootstrap.min.css">
+    <link rel="stylesheet" href="<?php echo $GLOBALS['assets_static_relative'] ?>/qtip2-2-2-1/jquery.qtip.min.css" />
+    <link rel="stylesheet" href="<?php echo $GLOBALS['assets_static_relative'] ?>/font-awesome-4-6-3/css/font-awesome.min.css">
+    <link rel="stylesheet" href="<?php echo $GLOBALS['assets_static_relative'] ?>/pure-0-5-0/pure-min.css">
+    <link rel="stylesheet" href="<?php echo $GLOBALS['web_root']; ?>/library/css/bootstrap_navbar.css" type="text/css">
+    <link rel="stylesheet" href="<?php echo $webroot; ?>/interface/main/messages/css/reminder_style.css?v=<?php echo rand(0); ?>" type="text/css">
+
+    <link rel="shortcut icon" href="<?php echo $webroot; ?>/sites/default/favicon.ico" />
+
+    <script type="text/javascript" src="<?php echo $GLOBALS['web_root']; ?>/library/dialog.js"></script>
+    <script type="text/javascript" src="<?php echo $GLOBALS['web_root']; ?>/library/textformat.js"></script>
```

Comment:
use cache busting mechanism in above 2 files:
``````
<script type="text/javascript" src="<?php echo $GLOBALS['web_root']; ?>/library/dialog.js?v=<?php echo $v_js_includes; ?>"></script>
<script type="text/javascript" src="<?php echo $GLOBALS['web_root']; ?>/library/textformat.js?v=<?php echo $v_js_includes; ?>"></script>
``````